--- SessionDelegate.swifts	2024-04-11 10:32:10
+++ SessionDelegate.swift	2024-04-11 10:32:26
@@ -343,6 +343,15 @@
             return
         }
 
+        let sizeString = ((downloadTask.response as? HTTPURLResponse)?.allHeaderFields["X-Uncompressed-Content-Length"]) as? String
+        if (sizeString != nil) {
+            let size = Int64(sizeString!)
+            downloadRequest.updateDownloadProgress(bytesWritten: bytesWritten,
+                                                   totalBytesExpectedToWrite: size!)
+            return
+            
+        }
+
         downloadRequest.updateDownloadProgress(bytesWritten: bytesWritten,
                                                totalBytesExpectedToWrite: totalBytesExpectedToWrite)
     }
