diff --git a/Pods/Alamofire/Source/Core/SessionDelegate.swift b/Pods/Alamofire/Source/Core/SessionDelegate.swift
index de0fcbb..5ebacfa 100644
--- a/Pods/Alamofire/Source/Core/SessionDelegate.swift
+++ b/Pods/Alamofire/Source/Core/SessionDelegate.swift
@@ -343,8 +343,15 @@ extension SessionDelegate: URLSessionDownloadDelegate {
             return
         }
 
-        downloadRequest.updateDownloadProgress(bytesWritten: bytesWritten,
+        let sizeString = ((downloadTask.response as? HTTPURLResponse)?.allHeaderFields["X-Uncompressed-Content-Length"]) as? String
+        if (sizeString != nil) {
+            let size = Int64(sizeString!)
+            downloadRequest.updateDownloadProgress(bytesWritten: bytesWritten,
                                                totalBytesExpectedToWrite: totalBytesExpectedToWrite)
+        } else {
+            downloadRequest.updateDownloadProgress(bytesWritten: bytesWritten,
+                                                   totalBytesExpectedToWrite: totalBytesExpectedToWrite)
+        }
     }
 
     open func urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didFinishDownloadingTo location: URL) {
